app/src/main/java/com/github/sdpsharelook/storage/RTDBWordListRepository.kt

```diff
@@ -1,55 +1,71 @@
 package com.github.sdpsharelook.storage
 
+import com.github.sdpsharelook.Word
+import com.github.sdpsharelook.authorization.AuthProvider
 import com.google.firebase.database.*
 import com.google.firebase.database.ktx.getValue
+import com.google.gson.Gson
 import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.channels.trySendBlocking
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.callbackFlow
 import kotlinx.coroutines.tasks.await
-import java.lang.UnsupportedOperationException
 import javax.inject.Inject
 
 class RTDBWordListRepository @Inject constructor(
-    private val firebaseDatabase: FirebaseDatabase
-) : IRepository<List<@JvmSuppressWildcards String>> {
+    private val firebaseDatabase: FirebaseDatabase,
+    private val auth: AuthProvider
+) : IRepository<List<@JvmSuppressWildcards Word>> {
 
     private val reference: DatabaseReference by lazy { firebaseDatabase.reference.child("wordlists") }
 
+
     /**
      * Gets an asynchronous data stream any updated [List] of [String]s
      *
      * @param name format: "path/name"
      * @return [Flow] of changes in the database at [name]
      */
-    override fun flow(name: String): Flow<Result<List<String>?>> = callbackFlow {
-        val fireListener = object : ChildEventListener {
-            override fun onChildAdded(snapshot: DataSnapshot, previousChildName: String?) {
-                val list = listOfNotNull(snapshot.getValue<String>())
-                trySendBlocking(Result.success(list))
-            }
+    override fun flow(name: String): Flow<Result<List<Word>?>> =
+        callbackFlow {
+            val fireListener = object : ChildEventListener {
+                val wordList = mutableListOf<Word>()
+                override fun onChildAdded(snapshot: DataSnapshot, previousChildName: String?) {
+                    val word = Gson().fromJson(snapshot.value.toString(), Word::class.java)
+                    wordList.add(wordList.size, word)
+                    trySendBlocking(Result.success(wordList))
+                }
+
+                override fun onChildChanged(snapshot: DataSnapshot, previousChildName: String?) {
+                    val list = listOfNotNull(snapshot.getValue<Word>())
+                    trySendBlocking(Result.success(list))
+                }
+
+                override fun onChildRemoved(snapshot: DataSnapshot) {
+                    val list: List<Word> = listOfNotNull(snapshot.getValue<Word>())
+                    trySendBlocking(Result.success(list))
+                }
```

Comment:
This is very cute, but nothing in your flow alows to know the difference between a ChildAdded, ChildChanged or ChildRemoved event